ServerSide/yoty/Core/Managers/Buyers/BuyersManager.cs

```diff
@@ -46,57 +47,56 @@ public async Task<Response> DeleteBuyer(string buyerId)
             {
-                return new Response<BuyerDTO>() {DTOObject = null, IsOperationSucceeded = false, SuccessOrFailureMessage = BuyerNotFoundFailString };
+                return new Response<BidsDTO>() { DTOObject = null, IsOperationSucceeded = false, SuccessOrFailureMessage = BuyerNotFoundFailString };
             }
-            BuyerDTO buyerDTO = _mapper.Map<BuyerDTO>(buyer);
-            return new Response<BuyerDTO>() { DTOObject = buyerDTO, IsOperationSucceeded = true, SuccessOrFailureMessage = this.getSuccessMessage() };
+
+            List<BidDTO> liveBids = buyer.CurrentParticipancies.Select(p => p.Bid).Where(bid=> FilterBuyerBids(bid, timeFiler)).Select(bid => _mapper.Map<BidDTO>(bid)).ToList();
+            return new Response<BidsDTO>() { DTOObject = BidsDTO.CreateDefaultBidsPage(liveBids), IsOperationSucceeded = true, SuccessOrFailureMessage = this.getSuccessMessage() };
         }
 
-        public async Task<Response<BidsDTO>> GetBuyerBids(string buyerId)
+
+        public async Task<Response<BidsDTO>> GetBidsCreatedByBuyer(string buyerId, BidsTime timeFiler)
         {
+            //validate existence
             var buyer = await _context.Buyers.Where(b => b.Id == buyerId).Include(b => b.CurrentParticipancies).ThenInclude(p => p.Bid).FirstOrDefaultAsync().ConfigureAwait(false);
             if (buyer == null)
             {
                 return new Response<BidsDTO>() { DTOObject = null, IsOperationSucceeded = false, SuccessOrFailureMessage = BuyerNotFoundFailString };
             }
-            List<BidDTO> liveBids = buyer.CurrentParticipancies.Select(p => p.Bid).Select(bid => _mapper.Map<BidDTO>(bid)).ToList();
-            return new Response<BidsDTO>() { DTOObject = BidsDTO.CreateDefaultBidsPage(liveBids), IsOperationSucceeded = true, SuccessOrFailureMessage = this.getSuccessMessage() };
+
+            var ownedBids = _context.Bids.Where(b => b.OwnerId == buyerId).Select(bid => _mapper.Map<BidDTO>(bid)).ToList();
+            return new Response<BidsDTO>() { DTOObject = BidsDTO.CreateDefaultBidsPage(ownedBids), IsOperationSucceeded = true, SuccessOrFailureMessage = this.getSuccessMessage() };
         }
 
-        public async Task<Response<BidsDTO>> GetBuyerLiveBids(string buyerId)
+        private static bool FilterBuyerBids(BidEntity buyerBid, BidsTime timeFiler)
         {
-            DateTime current_time = DateTime.Now;
-            var buyer = await _context.Buyers.Where(b => b.Id == buyerId).Include(b => b.CurrentParticipancies).ThenInclude(p => p.Bid).FirstOrDefaultAsync().ConfigureAwait(false);
-            if (buyer == null)
+            DateTime currentTime = DateTime.Now;
+
+            switch (timeFiler)
             {
-                return new Response<BidsDTO>() { DTOObject = null, IsOperationSucceeded = false, SuccessOrFailureMessage = BuyerNotFoundFailString };
+                case BidsTime.Old:
+                    return buyerBid.ExpirationDate <= currentTime;
```

Comment:
please add a TODO to change this later (I think "Old" should mean not active, as in Cancelled/Completed and after expiration date there is still Payment and Voting phase) after Phase will be added to bid we should use it instead of the exp date

Reply:
for now it's ok but I don't want us to forget